### PR TITLE
Refactor BDC dashboard to use ModelView Dashboard

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Main changes made :

1. Use ModelViewDashboard component for main dashboard
2. Each Service page is now a TabbedPanel component itself
3. PropertiesContainer component used for overview properties

The rest of the content is unchanged as that was already using base ModelView components. 

Old View : 

image

New View :

image

### CODE REVIEW

File: extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardOverviewPage.ts [hunk 1]
```diff
@@ -40,16 +31,23 @@ export class BdcDashboardOverviewPage extends BdcDashboardPage {
...
 		this.model.onDidUpdateBdcStatus(bdcStatus => this.eventuallyRunOnInitialized(() => this.handleBdcStatusUpdate(bdcStatus)));
 		this.model.onBdcError(error => this.eventuallyRunOnInitialized(() => this.handleBdcError(error)));
 	}
 
-	public create(view: azdata.ModelView): azdata.FlexContainer {
-		this.modelBuilder = view.modelBuilder;
-		const rootContainer = view.modelBuilder.flexContainer().withLayout(
+	public get container(): azdata.FlexContainer {
+		// Lazily create the container only when needed
+		//if (!this.rootContainer) {
```
Comment (alanrenmsft) [inline extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardOverviewPage.ts:43]: remove the commented out code?

File: extensions/big-data-cluster/src/bigDataCluster/dialog/intializingComponent.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Deferred } from '../../common/promise';
+
+export abstract class InitializingComponent {
```
Comment (alanrenmsft) [inline extensions/big-data-cluster/src/bigDataCluster/dialog/intializingComponent.ts:8]: do you see this class maybe useful for others as well? candidate for azdata namespace?

**Comment by alanrenmsft** [comment]
i assume you are tracking the link contrast ratio issue elsewhere, right?

**Comment by alanrenmsft** [comment]
looks great, it is amazing to see how much the code can be simplified.

File: extensions/big-data-cluster/src/bigDataCluster/dialog/intializingComponent.ts (hunk 2 above)
Comment (Charles-Gagnon) [reply extensions/big-data-cluster/src/bigDataCluster/dialog/intializingComponent.ts:8]: Hm, I'll think about that. Currently we're only really using it in the more complicated ModelView stuff like dashboards so it isn't too widely spread but yeah maybe useful for everyone to have easy access to it. I'll keep it in mind as we make more stuff whether we need it more commonly available.

**Comment by Charles-Gagnon** [comment]
@alanrenmsft https://github.com/microsoft/azuredatastudio/issues/7400

I'm going to look into why the hyperlink component isn't just using the normal hyperlink styling - but yeah that'll be done separately.

**Pull request merged**